docs/tutorials/kubernetes/data-sharing-with-vineyard-on-kubernetes.rst

```diff
@@ -0,0 +1,270 @@
+    vineyardd-sample-etcd-0             1/1     Running   0          3m53s   10.244.1.16   kind-worker3   <none>           <none>
+
+Assume we have two pods, one is a producer and the other is a consumer. 
+
+The producer yaml file is as follows.
+
+.. code:: bash
+
+    $ cat << EOF >> producer.yaml
+    apiVersion: apps/v1
+    kind: Deployment
+    metadata:
+      name: producer
+      namespace: vineyard-test
+    spec:
+      selector:
+        matchLabels:
+          app: producer
+      replicas: 1
+      template:
+        metadata:
+          labels:
+            app: producer
+        spec:
+          containers:
+          - name: producer
+            image: python:3.10
+            command:
+            - bash
+            - -c
+            - |
+              pip install vineyard numpy pandas > /dev/null 2>&1;
+              cat << EOF >> producer.py
+              import vineyard
+              import numpy as np
+              import pandas as pd
+              client = vineyard.connect()
+              client.put(pd.DataFrame(np.random.randn(100, 4), columns=list('ABCD')), persist=True, name="test_dataframe")
+              client.put((1, 1.2345, 'xxxxabcd'), persist=True, name="test_basic_data_unit");
+              client.close()
+              EOF
+              python producer.py;
+              sleep infinity;
+    EOF
+
+The consumer yaml file is as follows.
+
+.. code:: bash
+
+    $ cat << EOF >> consumer.yaml
+    apiVersion: apps/v1
+    kind: Deployment
+    metadata:
+      name: consumer
+      namespace: vineyard-test
+    spec:
+      selector:
+        matchLabels:
+          app: consumer
+      replicas: 1
+      template:
+        metadata:
+          labels:
+            app: consumer
+        spec:
+          containers:
+          - name: consumer
+            image: python:3.10
+            command:
+            - bash
+            - -c
+            - |
+              pip install vineyard numpy pandas > /dev/null 2>&1;
+              cat << EOF >> consumer.py
+              import vineyard
+              client = vineyard.connect()
+              dataframe_obj = client.get_name("test_dataframe")
+              print(client.get(dataframe_obj,fetch=True).sum())
+              unit_obj = client.get_name("test_basic_data_unit")
+              print(client.get(unit_obj,fetch=True))
+              client.close()
+              EOF
+              python consumer.py;
+              sleep infinity;
+    EOF
+
+Use the `vineyardctl` to schedule the two workloads on the vineyard cluster.
+
+.. code:: bash
+
+    # schedule the producer workload to the vineyard cluster and apply it to the kubernetes cluster
+    $ vineyardctl schedule workload -f producer.yaml --vineyardd-name vineyardd-sample \
+    --vineyardd-namespace vineyard-test -o yaml | kubectl apply -f -
+    deployment.apps/producer created
```

Comment:
For multiple-line commands, leave proper indentation, e.g., 


```
.. code:: bash
    # schedule the producer workload to the vineyard cluster and apply it to the kubernetes cluster
    $ vineyardctl schedule workload -f producer.yaml --vineyardd-name vineyardd-sample \
          --vineyardd-namespace vineyard-test -o yaml | kubectl apply -f -
          deployment.apps/producer created
```

Would be more readable.

Reply:
Get it. Thanks!